src/Page/Editor.purs

```diff
@@ -146,57 +132,84 @@ newtype EditorFields r f = EditorFields (r
   ))
 derive instance newtypeEditorFields :: Newtype (EditorFields r f) _
 
-formProxy :: F.FormProxy EditorFields
-formProxy = F.FormProxy
-
-proxies :: F.SProxies EditorFields
-proxies = F.mkSProxies formProxy
-
-validators :: forall m. Monad m => EditorFields Record (F.Validation EditorFields m)
-validators = EditorFields
-  { title: V.required >>> V.minLength 1
-  , description: V.required >>> V.minLength 1
-  , body: V.required >>> V.minLength 3
-  , tagList: F.hoistFn_ (map unwrap)
+type FormChildSlots = 
+  ( tagInput :: H.Slot (Const Void) TagInput.Message Unit )
+
+data FormAction 
+  = HandleTagInput TagInput.Message
+
+formInput :: forall m. Monad m => F.Input' EditorFields m
+formInput = 
+  { validators: EditorFields
+      { title: V.required >>> V.minLength 1
+      , description: V.required >>> V.minLength 1
+      , body: V.required >>> V.minLength 3
+      , tagList: F.hoistFn_ (map unwrap)
+      }
+  , initialInputs: Nothing 
   }
 
-renderFormless 
+formSpec 
   :: forall m
    . MonadAff m 
   => Maybe ArticleWithMetadata 
-  -> F.State EditorFields m 
-  -> F.HTML Query TagInput.Query Unit EditorFields m
-renderFormless mbArticle fstate =
-  HH.form_
-    [ HH.fieldset_
-      [ title
-      , description
-      , body
-      , HH.slot unit TagInput.component unit (HE.input $ F.raise <<< H.action <<< HandleTagInput)
-      , Field.submit $ if isJust mbArticle then "Commit changes" else "Publish"
-      ]
-    ]
+  -> F.Spec EditorFields () (Const Void) FormAction FormChildSlots Article m
```

Comment:
The Formless component now takes a `Spec`, which describes all the ways in which you'd like to override or extend the component. At minimum you will want to provide your own render function to render out your form fields. That could be done with just:

```
formSpec = F.defaultSpec { render  = render }
  where
  render { form } = ...
```

However, you can extend a form component in many more ways, represented in the `Spec` type above. In order, you should provide:

- the form type the form is running (here, `EditorFields`)
- any extra fields you want present in `State` (here, none, represented by `()`)
- any additional queries you want the component to be able to run (here, none, represented by `(Const Void)`)
- any additional actions you want the component to be able to run (here, `FormAction`, which handles messages from the `TagInput` component)
- any child components you plan to mount in the render function in the component (here, `FormChildSlots`, which describes the `TagInput` component)
- what messages the component will output, which you can define (here, a valid `Article` will be produced on successful form submission)
- what monad the component will run in (here, any monad supporting `MonadAff` -- including `AppM`)

If you choose to extend the component's query or action types, then you'll also want to write handlers for them: what should the component do when your query or action is triggered? That can be done by providing a `handleAction` or `handleQuery` function in the spec alongside your provided render function. 

Below, we've also intercepted the Formless `Submitted` message with `handleMessage`. There are more ways you can extend the spec, like hooking in to the `receive`, `initialize`, and `finalize` events; if you need that advanced behavior, please see the Formless library.

Most simple forms can just override the `render` function and possibly provide their own messages with `handleMessage`.